sympy/solvers/solveset.py

```diff
@@ -978,6 +985,319 @@ def _solveset(f, symbol, domain, _check=False):
+
+    `transolve` is designed in such a way that it becomes an easy task to
+    add a new class of equation solver.
+
+    The first task that needs to be done for adding your own solver
+    is to decide from where its `identificaion helper` will be
+    invoked. To do so determine the general form of the class of the
+    equation to place the invocation of the identification helper to
+    an appropriate place within transolve, for example the general
+    form of the exponential equations is `a**f(x) + b**g(x)` so we
+    need to place the invocation condition inside `is_Add` case. Once
+    the place for identification helper is determined, you can add a
+    call to solving helper. For your class of equation you need to
+    define your own identification and solving helpers. The
+    identification helper should be implemented for generalised cases
+    and should return either `True` if the given equation belongs to the
+    class otherwise `False`. Solving helpers needs to be implemented with
+    such heuristics or algorithms that solves most of the equations
+    belonging to that class. The value returned from this helper should
+    be handled properly in the main (transolve) function itself. It could
+    be either the exact solution or a reduced form of the equation which
+    `solveset` can handle.
+
+    Apart from this, few other things needs to be taken care while adding
+    an equation solver:
+
+    - Naming conventions
+      Name of the identification helper should be like  `_check_class`,
+      where `class` will be the name or abbreviation of the class of
+      equation. The solving helper will be named as `_class_solver`.
+      For eg: for exponential equation it becomes `_check_expo` and
+      `_expo_solver_`.
+    - Be sure to to consider corner cases
+    - Adding tests for each method.
+    - Adding a docstring to your helper that describes the method
+      implemented with its proof of correctness by explaining with few
+      examples. Since most of the methods devised to identify and solve
+      particular class of equations are mostly heuristics therefore it
+      becomes necessary to provide proof of correctness of these methods.
+      Following things needs to be included while writing the
+      documentation for the helpers:
+        - What is the purpose of the helper.
+        - How it solves the equation and how does it identifies the equation.
+        - Examples should be included to prove its correctness.
+        - What are the input parameters and what does the helper returns.
+
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import transolve
+    >>> from sympy import symbols, S
+
+    >>> x = symbols('x', real=True)
+    >>> transolve(5**(x-3) - 3**(2*x + 1), x, S.Reals)
+    {-log(375)/(-log(5) + 2*log(3))}
+    """
+
+    if f in flags.setdefault('tsolve_saw', []):
+        return ConditionSet(symbol, Eq(f, 0), domain)
+    flags['tsolve_saw'].append(f)
```

Comment:
I don't see any significance of this parameter as of now, neither from the code nor from the tests.

The following would work as well:

```diff
diff --git a/sympy/solvers/solveset.py b/sympy/solvers/solveset.py
index b1e24fe..78ea5bd 100644
--- a/sympy/solvers/solveset.py
+++ b/sympy/solvers/solveset.py
@@ -1066,7 +1066,7 @@ def _check_expo(f, symbol):
     return any(_ispow(i) for i in (ad, bd))


-def transolve(f, symbol, domain, **flags):
+def transolve(f, symbol, domain):
     """
     Function to solve transcendental equations. It is a helper to
     solveset and should be used internally as of now. `transolve`
@@ -1254,10 +1254,6 @@ class otherwise `False`. Solving helpers needs to be implemented with
     {-log(375)/(-log(5) + 2*log(3))}
     """

-    if f in flags.setdefault('tsolve_saw', []):
-        return ConditionSet(symbol, Eq(f, 0), domain)
-    flags['tsolve_saw'].append(f)
-
     # invert_complex handles the call to the desired inverter based
     # on the domain specified.
     lhs, rhs_s = invert_complex(f, 0, symbol, domain)
@@ -1286,7 +1282,7 @@ class otherwise `False`. Solving helpers needs to be implemented with
                 result = _solveset(new_f, symbol, domain)

             else:
-                result = transolve(f, symbol, domain, **flags)
+                result = ConditionSet(symbol, Eq(f, 0), domain)

         elif lhs.is_Pow:
             new_f = _expo_solver(lhs - rhs)
```

Reply:
We can remove the condition completely still it will return `ConditionSet` as result is initialised to `ConditionSet`.